Python/Product/TestAdapter/TestContainerDiscovererWorkspace.cs

```diff
@@ -173,7 +176,9 @@ class TestContainerDiscovererWorkspace : ITestContainerDiscoverer, IDisposable {
                 _testFilesUpdateWatcher.AddDirectoryWatch(workspace.Location);
                 oldWatcher?.Dispose();
 
-                var files = Directory.EnumerateFiles(workspace.Location, "*.*", SearchOption.AllDirectories);
+                var files = Directory.EnumerateFiles(workspace.Location, "*.py", SearchOption.AllDirectories)
+                    .Where(x => !IsFileExcluded(projInfo, x)
+                );
```

Comment:
Did you consider using EnumerateDirectories and checking for InterpreterPath? 
Maybe a more common case that enivronments folder is one of the folders directory under workspace.
then calling EnumerateFiles on the remaining folders.. 
I would assume its a faster then checking every file.

Reply:
Updated